optuna/study/study.py

```diff
@@ -896,6 +896,12 @@ def objective(trial):
 
         trial._validate()
 
+        if trial.values is not None and len(self.directions) != len(trial.values):
+            raise ValueError(
+                f"Although the number of objectives is {len(self.directions)}, "
+                f"the trial has {len(trial.values)} length values."
```

Comment:
```suggestion
                f"The added trial has {len(trial.values)} values, which is different from the number of objectives {len(self.directions)} in the study (determined by Study.directions)."
```
Nit: How about making the error message like this?